x/metoken/README.md

```diff
@@ -0,0 +1,326 @@
+6. **[Events](#events)**
+7. **[Parameters](#params)**
+8. **[EndBlock](#end-block)**
+
+## Concepts
+
+### Accepted Assets
+
+At the foundation of the `metoken` module is the _Index Registry_, which contains a list of Indexes with their meTokens, accepted assets and other parameters.
+
+This list is controlled by governance. Assets that are not in the index registry are not available for swapping or redeeming for the Index meToken.
+
+In an extreme case an asset can be deleted from the _Index Registry_, this will disallow to swap given asset for the Index meToken or redeem a meToken for that asset.
+
+### Index Parameters
+
+The Index will have the following parameters:
+
+- MeToken denom: the denomination of the Index meToken that will be given to user in exchange for accepted assets.
+- MeToken max supply: the maximum amount of Index meTokens that can be minted. A swap that requires to mint more
+  Index meToken than this value will result in an error.
+- Fee:
+  - Min fee: the minimum fee to be charged to the user. The applied fee will tend to decrease down to this value,
+    when the accepted asset is undersupplied in the index. It must be less than Balanced and Max fees. Valid values
+    0-∞.
+  - Balanced fee: the fee to be charged to the user when the index is balanced. It must be greater than Min fee and
+    lower than Max fee, it cannot be 0.
+  - Max fee: the maximum fee to be charged to the user. The applied fee will tend to increase up to this value,
+    when the accepted asset is oversupplied in the index. It must be greater than Min and Balanced fee. If the value
+    is 0, no Max fee will be applied.
+- Accepted Assets: a list where each asset will have the following parameters:
+  - Asset denom: the denomination of the underlying asset.
+  - Reserve portion: the portion of swapped assets that will be transferred to `metoken` module as reserves, and the
+    minimum portion that will be taken from `metoken` module reserves when a redemption occurs.
+  - Target allocation: the portion of an accepted asset the Index is targeting to have. This value will be
+    calculated in the addition and the update of an Index, as well as when an asset is deleted from it. The value
+    would be 1/amount of assets in the Index.
+
+### Swapping and Redeeming
+
+Users have the following actions available to them:
+
+- Swap accepted asset for Index meToken. Every accepted asset can be swapped for the Index meToken with 1:1 exchange
+  rate. The user will need to pay a [Dynamic Fee](#dynamic-fee) for the swap. The fee will be charged in the
+  accepted asset the Index meToken is swapped for.
+  
+  Index meToken amount needed for the swap will be minted and transferred to the user's account, while the accepted
+  asset for the swap will be transferred to the `leverage` module pools and the `metoken` module reserves.
+  The portion to be transferred to each one is determined by the _Index Registry_ configuration of each accepted asset.
```

Comment:
what happens when an asset has reached its max supply cap on umee?

Reply:
If the `leverage` module has reached the max supply cap for an asset we are trying to send to it, we will return an error. I think, since we will be adding the liquidity to `leverage` module, it will be a good idea to increase the `leverage` module max supply cap for every asset we add to the index for the same amount that balanced index should have.

So let's say we create a meUSD index with max supply cap of 1.800.000 meUSD tokens composed of USDC, USDT and IST. In that case we increase the max supply cap of USDC, USDT and IST Tokens by 600.000 each one in the `leverage` module.

cc: @toteki what do you think?

Reply:
Actually, this approach will also help to maintain the Index balanced when the `leverage` module is close to the max supply cap of an asset.

Reply:
1. I will document this, but I think there is a caveat. If the supply of an asset is limited by the `x/leverage` constraints we might never reach the `target_allocation` of a certain asset. So basically `x/leverage` supply of an asset controls the price of the fee in the Index, which I think is a bit unhealthy.